sysdig/resource_sysdig_secure_notification_channel_msteams.go

```
@@ -0,0 +1,213 @@
+	timeout := 5 * time.Minute
+
+	return &schema.Resource{
+		CreateContext: resourceSysdigSecureNotificationChannelMSTeamsCreate,
+		UpdateContext: resourceSysdigSecureNotificationChannelMSTeamsUpdate,
+		ReadContext:   resourceSysdigSecureNotificationChannelMSTeamsRead,
+		DeleteContext: resourceSysdigSecureNotificationChannelMSTeamsDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(timeout),
+			Update: schema.DefaultTimeout(timeout),
+			Read:   schema.DefaultTimeout(timeout),
+			Delete: schema.DefaultTimeout(timeout),
+		},
+
+		Schema: createSecureNotificationChannelSchema(map[string]*schema.Schema{
+			"url": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"template_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		}),
+	}
+}
+
+func resourceSysdigSecureNotificationChannelMSTeamsCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := getSecureNotificationChannelClient(meta.(SysdigClients))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	teamID, err := client.CurrentTeamID(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	notificationChannel, err := secureNotificationChannelMSTeamsFromResourceData(d, teamID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	notificationChannel, err = client.CreateNotificationChannel(ctx, notificationChannel)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(strconv.Itoa(notificationChannel.ID))
+	_ = d.Set("version", notificationChannel.Version)
+
+	return nil
+}
+
+func resourceSysdigSecureNotificationChannelMSTeamsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := getSecureNotificationChannelClient(meta.(SysdigClients))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, _ := strconv.Atoi(d.Id())
```

please do not ignore errors, here and also in other places